src/network/connection.cpp

```diff
@@ -2089,12 +1981,57 @@ void * ConnectionReceiveThread::Thread()
 	return NULL;
 }
 
+bool ConnectionReceiveThread::wait(int timeout_us, fd_set &readset) {
+	// Initialize the set
+	FD_ZERO(&readset);
+	int max_fd = 0;
+	for (std::vector<UDPSocket *>::const_iterator
```

Comment:
Shouldn't this use array-iteration syntax? (for (int i = 0; i != m_sockets.size(); i++) { m_sockets[i]->whatever(); ...